.ci/component_descriptor

```diff
@@ -1,3 +1,35 @@
 #!/usr/bin/env bash
 
+# Env variables provided by pipeline:
+# - BASE_DEFINITION_PATH: path to the basis component descriptor provided by the CI (contains only OCI images build by CI).
+# - CTF_PATH: pipeline will check for existing CTF archive and if exists, push the archive to a private repository.
+# - COMPONENT_DESCRIPTOR_PATH:
+#   - Filepath where the CI expects only the component-descriptor.yaml.
+#   - Cannot be used if with referenced resources in the component descriptor (like the Gardenlet blueprint).
+
+# call common hack script to enrich the base component descriptor
 "$(dirname $0)"/../hack/.ci/component_descriptor "$(dirname $0)"/..
+
+echo "Enriching component descriptor with blueprints of Gardener landscaper components"
+
+COMPONENT_ARCHIVE_PATH="$(mktemp -d)"
+
+# Common hack script has copied the enriched component descriptor to COMPONENT_DESCRIPTOR_PATH for backwards-compatibility.
+# If the descriptor exists at this filepath, it will be uploaded to a private registry.
+# However, this mechanism does not work if there are resource references in the component descriptor (like a local blueprint).
+# Instead, we have to build a CTF.
+# Therefore, move the existing component-descriptor.yaml to the component archive directory.
+mv "${COMPONENT_DESCRIPTOR_PATH}" "${COMPONENT_ARCHIVE_PATH}/component-descriptor.yaml"
+
+# further enrich the component descriptor with the blueprints of landscaper components
+component-cli component-archive resources add \
+"${COMPONENT_ARCHIVE_PATH}" \
+"$repo_root_dir/landscaper/resources.yaml"
```

Comment:
repo root is never defined.
```suggestion
"$(dirname $0)/../landscaper/resources.yaml"
```

Reply:
thanks for noticing. The component descriptor is now built and pushed correctly.